### ISSUE TITLE
Issue with EventPlayed EventHandler

### ISSUE THREAD

**Issue by MusicalMiqo** [melanchall/drywetmidi#56]
Consider the following code snippet:

```
playManager.EventPlayed += CCEventHandler;

private void CCEventHandler(object sender, MidiEventPlayedEventArgs e)
{
    var ccEvent = e.Event as ControlChangeEvent;
    if (ccEvent != null && ccEvent.ControlNumber == 15 && ccEvent.ControlValue == 0)
    {
        Console.WriteLine("Event triggered");
        playManager.MoveToStart();
    }
}
```

There is a single CC 15,0 event one bar before the end of my song, but when the playback manager reaches the CC event and jumps in time, it causes the playback manager to go into an infinite loop of playing the previous note (as a continuous, extremely loud tone).

I have tried changing the event to other types including SysEx events with the same result. I can't find any documentation nor sample code on the use of the EventPlayed event handler, so I am stumped for the moment.

**Comment by melanchall** [comment]
Thanks for the issue!

Can you provide the file to reproduce the problem?

**Comment by MusicalMiqo** [comment]
Yes, here is the sample midi as well as a test program I used to investigate the issue:
https://pastebin.com/xxQnVSZy
[Dream_repeat.zip](https://github.com/melanchall/drywetmidi/files/3990223/Dream_repeat.zip)

**Comment by melanchall** [comment]
Thanks for the file!

I've fixed the bug (commit 1207004a72c5849425effc9c05be8218de4d18f6). The fix is available in **develop** branch. Please tell me if the issue is resolved.

One small note. In your test program you use `while (playback.IsRunning)`. But `MoveToStart` stops playback while current time is changing, so the program will exit the loop. You should use separate variable to control whether playback is stopped or not.

**Comment by MusicalMiqo** [comment]
Thank you so much for the quick fix! The issue seems to be resolved now. Yes, the while loop was just for demo purposes, it is not in my production code.
